include/rocksdb/configurable.h

```diff
@@ -288,7 +288,7 @@ class Configurable {
  protected:
   // True once the object is prepared.  Once the object is prepared, only
   // mutable options can be configured.
-  std::atomic<bool> is_prepared_;
+  bool is_prepared_;
```

Comment:
I think there's more to this. I did some more digging and actually made the same change locally, then ran the unit test suite under TSAN. A handful of remote compaction related UTs were flagged due to multiple threads racing to update this variable for the `BytewiseComparator` Meyers singleton while deserializing `CompactionServiceInput` (which includes the column family options). I'm assuming the reason the flag got turned into an `atomic` in https://github.com/facebook/rocksdb/pull/8336 in the first place was to make these warnings go away. It did not, however, solve the root cause of the issue, namely that it is not safe to use the `Configurable` / `Customizable` framework from multiple threads (without synchronization) in the presence of "global" objects like `BytewiseComparator` (which is what the compaction service does).

Also Cc @jay-zhuang because this affects remote compaction (and might be related to the flaky test you were looking at).

Reply:
Having said that, I support ditching the `atomic` since it only gives the illusion of thread safety. The race affecting the compaction service could be handled in a separate PR I think.

Reply:
> The race affecting the compaction service could be handled in a separate PR I think.

Actually, scratch that part. The issue affects more than remote compaction, see e.g. how `DBTest2.MultiDBParallelOpenTest` under TSAN. I think it would be best if we fixed this in this PR.